unison-cli/src/Unison/Codebase/Editor/Slurp.hs

```diff
@@ -227,64 +201,36 @@ computeVarStatuses depMap varReferences codebaseNames =
+                  | otherwise -> Updated
+                _ -> Updated
             LD.TermReference {} ->
               case Set.toList existingTermsOrCtorsAtName of
-                [] -> DefOk New
-                rs | any Referent.isConstructor rs -> DefErr TermCtorCollision
+                [] -> New
+                rs | any Referent.isConstructor rs -> TermCtorCollision
                 [r]
-                  | LD.referent r == ld -> DefOk Duplicated
-                  | otherwise -> DefOk Updated
-                -- If there are many existing terms, they must be in conflict.
-                -- Currently we treat conflicts as errors rather than resolving them.
-                _ -> DefErr Conflict
+                  | LD.referent r == ld -> Duplicated
+                  | otherwise -> Updated
+                _ -> Updated
             LD.ConReference {} ->
               case Set.toList existingTermsOrCtorsAtName of
-                [] -> DefOk New
-                rs | any (not . Referent.isConstructor) rs -> DefErr CtorTermCollision
+                [] -> New
+                rs | any (not . Referent.isConstructor) rs -> CtorTermCollision
                 [r]
-                  | LD.referent r == ld -> DefOk Duplicated
-                  | otherwise -> DefOk Updated
-                -- If there are many existing terms, they must be in conflict.
-                -- Currently we treat conflicts as errors rather than resolving them.
-                _ -> DefErr Conflict
-
--- | Compute all definitions which can be added, or the reasons why a def can't be added.
-summarizeTransitiveStatus ::
-  forall v.
-  (Ord v, Show v) =>
-  Map (TaggedVar v) DefnStatus ->
-  Map (TaggedVar v) (Set (TaggedVar v)) ->
-  Map (TaggedVar v) (SummarizedStatus v)
-summarizeTransitiveStatus statuses deps =
-  flip imap (Align.align statuses deps) $ \tv -> \case
-    -- No dependencies
-    This selfStatus -> toSummary False selfStatus tv
-    That _set -> error $ "Encountered var without a status during slurping: " <> show tv
-    These selfStatus deps ->
-      let selfSummary = toSummary False selfStatus tv
-          summaryOfDeps = do
-            v <- Set.toList deps
-            depStatus <- maybeToList $ Map.lookup v statuses
-            pure $ toSummary True depStatus v
-       in foldl' pickPriorityStatus selfSummary summaryOfDeps
-  where
-    toSummary :: (Ord v, Show v) => Bool -> DefnStatus -> TaggedVar v -> SummarizedStatus v
-    toSummary isDep defNotes tv =
-      case defNotes of
-        DefOk Updated -> if isDep then DepStatus tv (DefOk Updated) else SelfStatus (DefOk Updated)
-        DefErr err ->
-          if isDep
-            then DepStatus tv (DefErr err)
-            else SelfStatus (DefErr err)
-        DefOk New -> SelfStatus (DefOk New)
-        DefOk Duplicated -> SelfStatus (DefOk Duplicated)
+                  | LD.referent r == ld -> Duplicated
+                  | otherwise -> Updated
```

Comment:
Maybe just delete this `otherwise` case and rely on the fall-through?

Otherwise maybe add a note to indicate that the lower `Updated` is for conflicts and this one is for "normal" updates.